app-misc/lcd4linux/lcd4linux-0.11.0_pre20170527-r2.ebuild

```diff
@@ -0,0 +1,186 @@
+	PICGraphic picoLCD picoLCDGraphic PNG PPM RouterBoard Sample SamsungSPF serdisplib ShuttleVFD
+	SimpleLCD T6963 TeakLCM Trefon ULA200 USBHUB USBLCD VNC WincorNixdorf X11
+)
+
+# Define the list of valid lcd4linux plugins.
+# Some plugins were removed from this list:
+# - imon: Uses telmond, which is only available on a fli4l router or an eisfair server.
+# - ppp: It has been replaced by the netdev plugin.
+# - seti: SETI@home software was replaced by sci-misc/boinc, which is not compatible.
+# - xmms: XMMS software was replaced by media-sound/xmms2, which is not compatible.
+IUSE_LCD4LINUX_PLUGINS=(
+	apm asterisk button_exec cpuinfo dbus diskstats dvb exec event
+	fifo file gps hddtemp huawei i2c_sensors iconv isdn kvv
+	loadavg meminfo mpd mpris_dbus mysql netdev netinfo pop3
+	proc_stat python qnaplog raspi sample statfs uname uptime
+	w1retap wireless
+)
+
+# Add supported drivers from 'IUSE_LCD_DEVICES' to 'IUSE' and 'REQUIRED_USE'
+REQUIRED_USE+=" || ( "
+for LCD_DEVICE in "${IUSE_LCD_DEVICES[@]}"; do
+	LCD_DEVICE=${LCD_DEVICE,,}
+	IUSE+=" lcd_devices_${LCD_DEVICE} "
+	REQUIRED_USE+=" lcd_devices_${LCD_DEVICE} "
+done
+REQUIRED_USE+=" ) "
+unset LCD_DEVICE
+
+# Add supported plugins from 'IUSE_LCD4LINUX_PLUGINS' to 'IUSE' and 'REQUIRED_USE'
+IUSE+=" ${IUSE_LCD4LINUX_PLUGINS[*]} "
+REQUIRED_USE+=" || ( ${IUSE_LCD4LINUX_PLUGINS[*]} ) "
+
+# Define dependencies for all drivers in 'IUSE_LCD_DEVICES'
+DEPEND_LCD_DEVICES="
+	lcd_devices_astusb? ( virtual/libusb:0= )
+	lcd_devices_bwct? ( virtual/libusb:0= )
+	lcd_devices_curses? ( sys-libs/ncurses:0= )
+	lcd_devices_dpf? ( virtual/libusb:0= )
+	lcd_devices_g15? ( virtual/libusb:0= )
+	lcd_devices_glcd2usb? ( virtual/libusb:0= )
+	lcd_devices_irlcd? ( virtual/libusb:0= )
+	lcd_devices_lcd2usb? ( virtual/libusb:0= )
+	lcd_devices_ledmatrix? ( virtual/libusb:0= )
+	lcd_devices_luise? (
+		dev-libs/luise-bin
+		virtual/libusb:0=
+	)
+	lcd_devices_matrixorbitalgx? ( virtual/libusb:0= )
+	lcd_devices_mdm166a? ( virtual/libusb:0= )
+	lcd_devices_picolcd? ( virtual/libusb:0= )
+	lcd_devices_picolcdgraphic? ( virtual/libusb:0= )
+	lcd_devices_png? (
+		media-libs/gd[png]
+		media-libs/libpng:0=
+	)
+	lcd_devices_ppm? ( media-libs/gd )
+	lcd_devices_samsungspf? ( virtual/libusb:0= )
+	lcd_devices_serdisplib? ( dev-libs/serdisplib )
+	lcd_devices_shuttlevfd? ( virtual/libusb:0= )
+	lcd_devices_trefon? ( virtual/libusb:0= )
+	lcd_devices_ula200? (
+		dev-embedded/libftdi:1=
+		virtual/libusb:0=
+	)
+	lcd_devices_usbhub? ( virtual/libusb:0= )
+	lcd_devices_usblcd? ( virtual/libusb:0= )
+	lcd_devices_vnc? ( net-libs/libvncserver )
+	lcd_devices_x11? ( x11-libs/libX11 )
+"
+
+# Define dependencies for all plugins in 'IUSE_LCD4LINUX_PLUGINS'
+DEPEND_LCD4LINUX_PLUGINS="
+	asterisk? ( net-misc/asterisk )
+	dbus? ( sys-apps/dbus )
+	gps? ( dev-libs/nmeap )
+	hddtemp? ( app-admin/hddtemp )
+	iconv? ( virtual/libiconv )
+	mpd? ( media-libs/libmpd )
+	mpris_dbus? ( sys-apps/dbus )
+	mysql? (
```

Comment:
Actually, this whole block is wrong. Those two libraries aren't exclusive, so the program can link only to one of them, in which case `||` is plain wrong. And if it has no explicit mariadb support, then it probably doesn't support it in the first place.

However, since this has nothing to do with the change in question, I'm going to merge this PR. Please submit a fix for mysql usage separately.

Reply:
Done.